Text/LaTeX/Base/Class.hs

```diff
@@ -40,32 +36,72 @@ import Data.Text (Text)
 
 -- | Map a 'LaTeX' value to its equivalent in any 'LaTeXC' instance.
 fromLaTeX :: LaTeXC l => LaTeX -> l
-fromLaTeX l = liftListL (\_ -> l) []
+fromLaTeX l = liftListL (const l) []
 
 -- | Lift a inner function of 'LaTeX' values into any 'LaTeXC' instance.
 liftL :: LaTeXC l => (LaTeX -> LaTeX) -> l -> l
-liftL f x = liftListL (\[x] -> f x) [x]
+liftL f x1 = liftListL (\[x1] -> f x1) [x1]
 
 -- | Variant of 'liftL' with a two arguments function.
 liftL2 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX) -> l -> l -> l
-liftL2 f x y = liftListL (\[x,y] -> f x y) [x,y]
+liftL2 f x1 x2 = liftListL (\[x1,x2] -> f x1 x2) [x1,x2]
 
 -- | Variant of 'liftL' with a three arguments function.
 liftL3 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX -> LaTeX) -> l -> l -> l -> l
-liftL3 f x y z = liftListL (\[x,y,z] -> f x y z) [x,y,z]
+liftL3 f x1 x2 x3 = liftListL (\[x1,x2,x3] -> f x1 x2 x3) [x1,x2,x3]
+
+-- | Variant of 'liftL' with a four arguments function.
+liftL4 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX) -> l -> l -> l -> l -> l
+liftL4 f x1 x2 x3 x4 = liftListL (\[x1,x2,x3,x4] -> f x1 x2 x3 x4) [x1,x2,x3,x4]
+
+-- | Variant of 'liftL' with a five arguments function.
+liftL5 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX) -> l -> l -> l -> l -> l -> l
+liftL5 f x1 x2 x3 x4 x5 = liftListL (\[x1,x2,x3,x4,x5] -> f x1 x2 x3 x4 x5) [x1,x2,x3,x4,x5]
+
+-- | Variant of 'liftL' with a six arguments function.
+liftL6 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX) -> l -> l -> l -> l -> l -> l -> l
+liftL6 f x1 x2 x3 x4 x5 x6 = liftListL (\[x1,x2,x3,x4,x5,x6] -> f x1 x2 x3 x4 x5 x6) [x1,x2,x3,x4,x5,x6]
+
+-- | Variant of 'liftL' with a seven arguments function.
+liftL7 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX) -> l -> l -> l -> l -> l -> l -> l -> l
+liftL7 f x1 x2 x3 x4 x5 x6 x7 = liftListL (\[x1,x2,x3,x4,x5,x6,x7] -> f x1 x2 x3 x4 x5 x6 x7) [x1,x2,x3,x4,x5,x6,x7]
+
+-- | Variant of 'liftL' with an eight arguments function.
+liftL8 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX) -> l -> l -> l -> l -> l -> l -> l -> l -> l
+liftL8 f x1 x2 x3 x4 x5 x6 x7 x8 = liftListL (\[x1,x2,x3,x4,x5,x6,x7,x8] -> f x1 x2 x3 x4 x5 x6 x7 x8) [x1,x2,x3,x4,x5,x6,x7,x8]
+
+-- | Variant of 'liftL' with a nine arguments function.
+liftL9 :: LaTeXC l => (LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX -> LaTeX) -> l -> l -> l -> l -> l -> l -> l -> l -> l -> l
+liftL9 f x1 x2 x3 x4 x5 x6 x7 x8 x9 = liftListL (\[x1,x2,x3,x4,x5,x6,x7,x8,x9] -> f x1 x2 x3 x4 x5 x6 x7 x8 x9) [x1,x2,x3,x4,x5,x6,x7,x8,x9]
+
+
+-- Map the given functor of LaTeX values to TeX arguments.
+fixArgs :: Functor f => f LaTeX -> f TeXArg
+fixArgs = fmap FixArg
+
+optFixArgs :: Int -> [LaTeX] -> [TeXArg]
+optFixArgs _ [] = []
+optFixArgs k xa@(x:xs) | k <= 0 = map FixArg xa
+                       | otherwise = OptArg x : optFixArgs (k-1) xs
+
```

Comment:
Could you please add Haddock documentation for the two new exported functions?

Thanks!